build_win_lnx.py

```diff
@@ -0,0 +1,176 @@
+
+    cts_solution_cmd_win = 'cmake ..\. -DCMAKE_BUILD_TYPE=release -DCL_INCLUDE_DIR=..\OpenCL-Headers -DCL_LIB_DIR=..\OpenCL-ICD-Loader\\build\\Release\\ -DCL_LIBCLCXX_DIR=..\libclcxx\\include -DCL_OFFLINE_COMPILER=..\\dummy\\path\\to\\compiler -DCMAKE_RUNTIME_OUTPUT_DIRECTORY=./bin -DOPENCL_LIBRARIES=OpenCL -DD3D10_IS_SUPPORTED=ON -DD3D11_IS_SUPPORTED=ON'
+
+    cts_solution_cmd_win_x64_msvc_2017 = cts_solution_cmd_win + ' -G "Visual Studio 15 2017" -A x64 -DCMAKE_CL_64=ON -DARCH=x86_64'
+    cts_solution_cmd_win_x64_icc_2017 = cts_solution_cmd_win_x64_msvc_2017 + ' -T"Intel C++ Compiler 19.0"'
+
+    cts_solution_cmd_win_x64_msvc_2019 = cts_solution_cmd_win + ' -G "Visual Studio 16 2019" -A x64 -DCMAKE_CL_64=ON -DARCH=x86_64'
+    cts_solution_cmd_win_x64_icc_2019 = cts_solution_cmd_win_x64_msvc_2019 + ' -T"Intel C++ Compiler 19.0"'
+
+    cts_solution_cmd_win_x86_msvc_2017 = cts_solution_cmd_win + ' -G "Visual Studio 15 2017" -DCMAKE_CL_64=OFF -DARCH=i686'
+    cts_solution_cmd_win_x86_icc_2017 = cts_solution_cmd_win_x86_msvc_2017 + ' -T"Intel C++ Compiler 19.0"'
+
+    cts_solution_cmd_win_x86_msvc_2019 = cts_solution_cmd_win + ' -G "Visual Studio 16 2019" -DCMAKE_CL_64=OFF -DARCH=i686'
+    cts_solution_cmd_win_x86_icc_2019 = cts_solution_cmd_win_x86_msvc_2019 + ' -T"Intel C++ Compiler 19.0"'
+
+    cts_solution_cmd_linux = 'cmake -DCL_INCLUDE_DIR=../OpenCL-Headers -DCL_LIB_DIR=../OpenCL-ICD-Loader/build -DCL_LIBCLCXX_DIR=../libclcxx -DCMAKE_RUNTIME_OUTPUT_DIRECTORY=./bin -DOPENCL_LIBRARIES="-lOpenCL -lpthread" ..'
+
+    if platform == "win32":
+        print("Windows operating system")
+        if bitness == "x64" and msvc_version == "2017":
+            icd_cmd = icd_solution_cmd_win_x64_msvc_2017
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x64_msvc_2017
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x64_icc_2017
+        elif bitness == "x64" and msvc_version == "2019":
+            icd_cmd = icd_solution_cmd_win_x64_msvc_2019
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x64_msvc_2019
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x64_icc_2019
+        elif bitness == "x86" and msvc_version == "2017":
+            icd_cmd = icd_solution_cmd_win_x86_msvc_2017
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x86_msvc_2017
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x86_icc_2017
+        elif bitness == "x86" and msvc_version == "2019":
+            icd_cmd = icd_solution_cmd_win_x86_msvc_2019
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x86_msvc_2019
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x86_icc_2019
+
+    elif platform[:5] == "linux":
+        print("Linux operating system")
+        compiler = "gcc"
```

Comment:
Is this always true, or could the compiler be `clang` if an environment var like `CC/CXX` is set?

Reply:
I've added only support for gcc. Feel free to extend support for any compiler you want

Reply:
Only supporting gcc is fine, was mostly concernced about the script printing `Compiler: gcc` when another compiler was actually being used by cmake. But now that message is guarded to windows only on line 117, so avoiding the potential for misleading output.